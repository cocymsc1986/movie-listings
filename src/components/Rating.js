import React, { Fragment } from 'react';

export const Rating = ({ ratingValue, updateScoreFilter }) => {
	return (
		<Fragment>
			<label htmlFor="rating"><h3>Min Rating ({ratingValue})</h3></label>
			<div className="filters__rating">
				<input id="rating" type="range" min="0" max="10" step="0.5" value={ratingValue} onChange={updateScoreFilter} />
			</div>
		</Fragment>
	);
};

export default Rating;