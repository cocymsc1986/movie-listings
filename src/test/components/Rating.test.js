import React from 'react';

import Rating from '../../components/Rating';

describe('Rating component', () => {
	test('Should render', () => {
		const wrapper = shallow(<Rating ratingValue={5} updateScoreFilter={() => {}} />);

		expect(wrapper).toMatchSnapshot();
	});
});