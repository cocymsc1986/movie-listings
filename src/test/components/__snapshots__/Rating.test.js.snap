// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating component Should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    ratingValue={5}
    updateScoreFilter={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <label
          htmlFor="rating"
>
          <h3>
                    Min Rating (
                    5
                    )
          </h3>
</label>,
        <div
          className="filters__rating"
>
          <input
                    id="rating"
                    max="10"
                    min="0"
                    onChange={[Function]}
                    step="0.5"
                    type="range"
                    value={5}
          />
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h3>
            Min Rating (
            5
            )
</h3>,
          "htmlFor": "rating",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Min Rating (",
              5,
              ")",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Min Rating (",
            5,
            ")",
          ],
          "type": "h3",
        },
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <input
            id="rating"
            max="10"
            min="0"
            onChange={[Function]}
            step="0.5"
            type="range"
            value={5}
/>,
          "className": "filters__rating",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "rating",
            "max": "10",
            "min": "0",
            "onChange": [Function],
            "step": "0.5",
            "type": "range",
            "value": 5,
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": "div",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <label
            htmlFor="rating"
>
            <h3>
                        Min Rating (
                        5
                        )
            </h3>
</label>,
          <div
            className="filters__rating"
>
            <input
                        id="rating"
                        max="10"
                        min="0"
                        onChange={[Function]}
                        step="0.5"
                        type="range"
                        value={5}
            />
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h3>
              Min Rating (
              5
              )
</h3>,
            "htmlFor": "rating",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Min Rating (",
                5,
                ")",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Min Rating (",
              5,
              ")",
            ],
            "type": "h3",
          },
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              id="rating"
              max="10"
              min="0"
              onChange={[Function]}
              step="0.5"
              type="range"
              value={5}
/>,
            "className": "filters__rating",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "rating",
              "max": "10",
              "min": "0",
              "onChange": [Function],
              "step": "0.5",
              "type": "range",
              "value": 5,
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
